Don't use -G, it results in a relocation error against _DYNAMIC on
amd64, use -shared instead per the manpage, and as it works as well.

--- configure.in
+++ configure.in
@@ -2481,7 +2481,7 @@
         AC_LANG_RESTORE
     else
        ASFLAGS="$ASFLAGS -fPIC"
-       DSO_LDOPTS='-G'
+       DSO_LDOPTS='-shared'
        _WARNINGS_CFLAGS=''
        _WARNINGS_CXXFLAGS=''
        if test "$OS_RELEASE" = "5.3"; then
